File: Java/Jąkała.java
import java.util.HashSet;
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int k = input.nextInt();
        input.nextLine();
        HashSet<Character> doubledLetters = new HashSet<>();
        String[] letters = input.nextLine().split(" ");
        for (String letter : letters)
            doubledLetters.add(letter.charAt(0));
        String word = input.nextLine();
        StringBuilder result = new StringBuilder();
        for (char c : word.toCharArray())
        {
            result.append(c);
            if (doubledLetters.contains(c))
                result.append(c);
        }
        System.out.print(result);
        input.close();
    }
}

File: C++/Jąkała.cpp
#include <string>
#include <iostream>
#include <unordered_set>

int main() 
{
    int k;
    std::cin >> k;
    char letter;
    std::unordered_set<char> doubled_letters;
    while(k--)
    {
        std::cin >> letter;
        doubled_letters.insert(letter);
    }
    std::string word, result;
    std::cin >> word;
    for (char c : word)
    {
        result += c;
        if (doubled_letters.count(c))
            result += c;
    }
    std::cout << result;
    return 0;
}